nemo/core/classes/modelPT.py

```diff
@@ -1782,16 +1785,46 @@ def _setup_profiling(self):
                 self._memory_profile_end_step = self.cfg.memory_profile.get('end_step', 0)
                 self._memory_profile_rank = self.cfg.memory_profile.get('rank', 0)
                 self._memory_profile_output_path = self.cfg.memory_profile.get('output_path', None)
+                self._memory_profile_snapshot_file_activation = (
+                    f'{self._memory_profile_output_path}/memory_profile_rank{self._memory_profile_rank}_act.pickle'
+                )
+                self._memory_profile_snapshot_file_weight = (
+                    f'{self._memory_profile_output_path}/memory_profile_rank{self._memory_profile_rank}_weight.pickle'
+                )
+                self._memory_profile_snapshot_file_oom = (
+                    f'{self._memory_profile_output_path}/memory_profile_rank{self._memory_profile_rank}_oom.pickle'
+                )
+                # CUDA memory profiling options: analysis
+                self._memory_profile_analysis_enabled = self.cfg.memory_profile.get('analysis_enabled', False)
+                self._memory_profile_analysis_path = os.path.join(self._memory_profile_output_path, f"analysis")
+                # CUDA memory profiling options: weight profile & OOM profile
+                self._memory_profile_weight_enabled = True  # Set as True
+                self._memory_profile_oom_enabled = True  # Set as True
+
+                if (
+                    self._memory_profile_weight_enabled
+                ):  # For the weight profile, we record the snapshot from the initialization, and dump it before batch-0 start. Note: can't use `get_rank()` during initialization, since nccl_init_group is not called yet.
+                    logging.info(
+                        f"====== Rank[{self._memory_profile_rank}], Initialization. Start CUDA memory profiling: Weight ======"
+                    )
+                    torch.cuda.memory._record_memory_history()
```

Comment:
Does this need specific version of pytorch ? If so please guard it so that it doesn't run on lower pytorch versions